src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,95 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Storage class represents storage object which interacts
+ * with the task data stored in the txt file.
+ */
+public class Storage {
+
+    /**
+     * Reads the txt file from Info/data.txt and returns
+     * an ArrayList of all the tasks.
+     *
+     * @return An ArrayList of the tasks read from the file.
+     * @throws FileNotFoundException If the txt file does not exist.
+     * @throws IOException If the file cannot be found by scanner or I/O error occurred.
+     */
+    public ArrayList<Task> readFile() throws IOException {
+        ArrayList<Task> botArray  = new ArrayList<>();
```

Comment:
Hmm... perhaps the name 'botArray' can be changed to something else more intuitive? The bot to me can seem like a "bottom array" or a "robot array".